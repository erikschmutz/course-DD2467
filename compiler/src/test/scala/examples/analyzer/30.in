type a = {
    x: int
};;

type b = {
    x: int
};;

let a = {
    x=1
};;

(*====================
type a : a {x:int}
type b : b {x:int}
val a : b {x:int}
- :b {x:int}
====================*)