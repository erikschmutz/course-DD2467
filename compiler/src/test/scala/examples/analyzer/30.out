AnalyzerResult(Record(b,List(RecordField(x,Integer())),List()),Enviroment(List(TypeEntry(Identifier(a),Record(a,List(RecordField(x,Integer())),List())),TypeEntry(Identifier(b),Record(b,List(RecordField(x,Integer())),List())),ValEntry(Identifier(a),Record(b,List(RecordField(x,Integer())),List())))))